src/commands/misc/role.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  ChatInputCommandInteraction,
+  GuildMember,
+  MessageReaction,
+  PartialMessageReaction,
+  PartialUser,
+  Role,
+  SlashCommandBuilder,
+  User,
+} from "discord.js";
+
+import { discordClient } from "../../clients/discord";
+
+const ROLES_CHANNEL = "741479910886866944";
+
+const THEY_THEM = "741479573337800706";
+const HE_HIM = "741479366319538226";
+const SHE_HER = "741479514902757416";
+const LISTENER = "466622497991688202";
+const NO_ALERTS = "512522219574919179";
+const LFG = "754473984661258391";
+
+const EMOJI_TO_ROLE = new Map([
+  ["🇹", THEY_THEM],
+  ["♀️", SHE_HER],
+  ["♂️", HE_HIM],
+  ["👂", LISTENER],
+  ["🚫", NO_ALERTS],
+  ["✅", LFG],
+]);
+
+export const data = new SlashCommandBuilder()
+  .setName("role")
+  .setDescription("Self-serve certain roles on the server")
+  .addSubcommand((subcommand) =>
+    subcommand
+      .setName("pronouns")
+      .setDescription("Please indicate your preferred pronouns")
+      .addStringOption((option) =>
+        option
+          .setName("pronoun")
+          .setDescription("Toggle a preferred pronoun. You can come back to set multiple")
+          .addChoices(
+            { name: "they/them", value: THEY_THEM },
+            { name: "he/him", value: HE_HIM },
+            { name: "she/her", value: SHE_HER }
+          )
+          .setRequired(true)
+      )
+  )
+  .addSubcommand((subcommand) =>
+    subcommand
+      .setName("alerts")
+      .setDescription("Let us know how you want to be notified on the server")
+      .addStringOption((option) =>
+        option
+          .setName("choice")
+          .setDescription("Select your perferred alerts. These are mutually exclusive")
+          .addChoices(
+            { name: "Listener (occasional alerts)", value: LISTENER },
+            { name: "No alerts (no alerrts)", value: NO_ALERTS }
+          )
+          .setRequired(true)
+      )
+  )
+  .addSubcommand((subcommand) =>
+    subcommand
+      .setName("gaming")
+      .setDescription(
+        "Let us know if you want to receive alerts when folks are looking for gaming parties"
```

Comment:
(actually, real comment here: the looking for games tag feels like useless cruft we don't really need on the server. fine to have oaf still run it just like what's the point lol)

Reply:
I was just trying to replicate the roles channel. Happy to remove after this if server admin-types agree.